tests/test_fips.py

```diff
@@ -20,39 +20,43 @@
 from bci_tester.data import CONTAINERS_WITH_ZYPPER
 from bci_tester.data import OS_VERSION
 from bci_tester.fips import FIPS_DIGESTS
-from bci_tester.fips import host_fips_enabled
 from bci_tester.fips import NONFIPS_DIGESTS
 
 
-# building the documentation will fail on a non-FIPS host otherwise
-if "sphinx" not in sys.modules:
-    assert (
-        host_fips_enabled()
-    ), "The host must run in FIPS mode for the FIPS test suite"
+#: Error message from OpenSSL when a non-FIPS digest is selected in FIPS mode
+FIPS_ERR_MSG = (
+    "not a known digest" if OS_VERSION == "15.3" else "Error setting digest"
+)
 
+pytestmark = pytest.mark.skipif(
+    OS_VERSION == "tumbleweed",
+    reason="no FIPS module in tumbleweed yet",
+)
 
 #: multistage :file:`Dockerfile` that builds the program from
 #: :py:const:`FIPS_TEST_DOT_C` using gcc and copies it, ``libcrypto``, ``libssl``
 #: and ``libz`` into the deployment image. The libraries must be copied, as they
 #: are not available in the minimal container images.
 DOCKERFILE = f"""FROM $builder as builder
 
+
 WORKDIR /src/
 COPY fips-test.c /src/
-RUN zypper -n ref && zypper -n in gcc libopenssl-devel && zypper -n clean
-RUN gcc -Og -g3 fips-test.c -Wall -Wextra -Wpedantic -lcrypto -lssl -o fips-test
+RUN zypper -n ref && zypper -n in gcc openssl libopenssl-devel && zypper -n clean
+RUN gcc -O2 fips-test.c -Wall -Werror -lcrypto -lssl -o fips-test
 
 FROM $runner
 
-COPY --from=builder /src/fips-test /bin/fips-test
-COPY --from=builder /usr/lib64/libcrypto.so.1.1 /usr/lib64/
-COPY --from=builder /usr/lib64/libssl.so.1.1 /usr/lib64/
-COPY --from=builder {'/usr' if OS_VERSION not in ('15.3', '15.4') else ''}/lib64/libz.so.1 /usr/lib64/
-COPY --from=builder /usr/lib64/engines-1.1 /usr/lib64/engines-1.1
-COPY --from=builder /usr/lib64/.libcrypto.so.1.1.hmac /usr/lib64/
-COPY --from=builder /usr/lib64/.libssl.so.1.1.hmac /usr/lib64/
+COPY --from=builder /src/fips-test /usr/local/bin/fips-test
+COPY --from=builder /usr/bin/openssl /usr/bin/openssl
+COPY --from=builder /usr/lib64/libcrypto.so.* /usr/lib64/
+COPY --from=builder /usr/lib64/libssl.so.* /usr/lib64/
+COPY --from=builder /usr/lib64/libz.so.1 /usr/lib64/
```

Comment:
This will break on 15.3 & 15.4:
```
❯ podman run --rm -it registry.suse.com/bci/bci-base:15.4 ls -al /usr/lib64/libz.so.1
ls: cannot access '/usr/lib64/libz.so.1': No such file or directory
```